feat(Geosuggest): Add `debounce` prop, default to 250ms

lsanwick opened the pull request:
I'm implementing debounce in a similar manner to Issue #131, with a default delay of 250ms.  The current implementation is actually unusable on IE11 because the cost in time to re-render on every keystroke causes the browser to eat keystrokes while you are typing.  Adding a debounce makes the component at least functional.

ro-ka commented:
Thanks for the PR! But I still think it makes sense to have this delay without a configuration. Do you mind to remove the prop from the component but keep the delay?

ro-ka commented on package.json:
```diff
@@ -17,7 +17,8 @@
   "dependencies": {
-    "classnames": "^2.2.3"
+    "classnames": "^2.2.3",
+    "lodash": "^4.13.1"
```
Please only add ![lodash.debounce](https://www.npmjs.com/package/lodash.debounce).

lsanwick commented:
Rebased, and moved to the specific lodash package, thanks for the pointer!

I'll still argue for having a parameter available to change the debounce time, just because individual users may have different needs.  Some may be willing to sacrifice query volume for a faster response time, while others might want to minimize the number of queries made.  I'm happy to rename the parameter to be something clearer to an end user, like `queryDelay`. 

Also, for IE11 users (a minority, I admit) the default delay is still too short to prevent buggy behavior for them, and I don't want the default to have to be something like 500ms.  I'll continue to work on trying to improve the performance in IE11 though.

ro-ka commented:
I think `queryDelay` is nicer. Can you update? Thanks!  :)

lsanwick commented:
Closing this in favor of #163 because this PR was coming off my master branch.

dimafeldman commented:
Seems like this queryDelay of 250ms, causes android browsers to sometimes trigger onChange event, after the suggest has already been selected. to reproduce:
1. open the [demo page](http://ubilabs.github.io/react-geosuggest/) on an android browser
2. start typing in this address: 56 Varet Street, Brooklyn
3. select the first result
4. you will see that the correct geosuggest is being selected
5. you will also see that the onChange event is being triggered once again, and basically invalidates your previous suggestion, and keeps the suggests dropdown open

I couldn't find the exact cause for that issue - it's very specific to android browser, and to overcome it, I changed the queryDelay to 0...

would love to hear your input :)

Thanks
